k8s/helm/values/production.yaml

```diff
@@ -0,0 +1,100 @@
+# Default values for prerelease.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 2
+revisionHistoryLimit: 3
+
+# Define container liveness and readiness checks
+# A Pod is considered "live" when it is able to respond to client requests.
+# A Pod is considered "ready" when it has completed initialization and should be one of the backends for a K8s Service resource.
+livenessProbe:
+  initialDelaySeconds: 120
+  periodSeconds: 30
+  failureThreshold: 6
+  http:
+    path: /alive
+readinessProbe:
+  initialDelaySeconds: 120
+  periodSeconds: 30
+  failureThreshold: 6
+  http:
+    path: /alive
+
+
+image:
+  repository: "harbor.devops.k8s.rcsb.org/rcsb/rcsb-chemsearch"
+  pullPolicy: Always
+  tag: "production"
+
+imagePullSecrets: "harbor-docker-registry-conf"
+
+deploymentStrategy:
+  #https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
+  #Type is either RollingUpdate or Recreate
+  type: "RollingUpdate"
+  #For rolling update, what percentage of total pods can be created above desired amount
+  maxSurge: 25%
+  #For rolling update, what percentage of total pods can be brought down to update
+  maxUnavailable: 25%
+
+podSecurityContext:
+  fsGroup: 1000
+  runAsNonRoot: true
+  runAsUser: 1000
+  runAsGroup: 1000
+  seccompProfile:
+    type: RuntimeDefault
+
+containerSecurityContext:
+  readOnlyRootFilesystem: true
+  allowPrivilegeEscalation: false
+
+service:
+  type: ClusterIP
+  port: 80
+  containerPort: 8000
+
+resources:
+  limits:
+    cpu: 256m
+    memory: 8Gi
+    ephemeral-storage: "9G"
+  requests:
+    cpu: 4m
+    memory: 1Gi
+    ephemeral-storage: "9G"
```

Comment:
The gap between the request and limit resources are pretty large. We should profile the performance of the application right now deployed in Kubernetes and in OpenStack to find the appropriate values, and ensure that those values are set appropriately.